templates/nginx-stream.conf.j2

```diff
@@ -0,0 +1,18 @@
+upstream restund {
+    least_conn;
+    zone backend 64k;
+    server {{ hostvars[inventory_hostname]['ansible_' + restund_network_interface].ipv4.address }}:{{ restund_tcp_listen_port }} weight=1 max_fails=1 fail_timeout=10s;
+}
+
+# see also https://docs.nginx.com/nginx/admin-guide/security-controls/terminating-ssl-tcp/
+server {
+    listen {{ restund_tls_listen_port }} ssl;
+    proxy_pass restund;
+    proxy_timeout 3s;
+    proxy_connect_timeout 1s;
+    proxy_protocol off;
+    ssl_certificate /etc/letsencrypt/live/{{ certbot_domain }}/fullchain.pem;
+    ssl_certificate_key /etc/letsencrypt/live/{{ certbot_domain }}/privkey.pem;
+
+    # TODO: which other settings (ciphers protocols etc) make sense here in the restund usecase (where we don't rely on the transport encryption of TLS, since the content is already end-to-end encrypted, and most calls go through regular UDP/TCP anyway - the TLS is used in networking setups where UDP and plain TCP is blocked, but TLS is allowed, i.e. a fallback mode only)
```

Comment:
I think the defaults look reasonable, all clients should support those ciphers/protocols (note that tls v1.3 is _not_ available, we may want to keep it in mind in case some clients stop supporting anything below that)

Reply:
The defaults set no ciphers nor any protocols, so even old SSL protocol etc is not denied. Is that really reasonable?

Reply:
Looking at: http://nginx.org/en/docs/stream/ngx_stream_ssl_module.html

Default: ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
Default: ssl_ciphers HIGH:!aNULL:!MD5;

That's why I hinted at the defaults being acceptable; unless we want to restrict the ciphers but since we don't really care about the actual content (it will preferably be sent over UDP or plain TCP) does not seem worth fiddling around with.

Reply:
Hm. I actually had a look at SSL labs with the defaults, which results in a B, and it doesn't support TLS 1.3, which is used by browsers these days. So it might be a problem to not support TLS 1.3. I switched to intermediate config (or parts of that) in 7de4011dabca2cb431b795db808719f5d4ab71e3

Fine also for you?